core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractReactorSinkProvider.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.processor.strategy;
+
+import static java.lang.Thread.currentThread;
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.api.transaction.Transaction;
+import org.mule.runtime.core.api.transaction.TransactionCoordination;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.RemovalListener;
+import reactor.core.publisher.FluxSink;
+
+public abstract class AbstractReactorSinkProvider implements ReactorSinkProvider {
```

Comment:
I would change the name to AbstractCachedThreadReactorSinkProvider and add a jdoc though this is not an API.